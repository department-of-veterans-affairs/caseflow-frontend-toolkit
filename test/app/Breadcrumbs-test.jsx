import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Breadcrumbs from '../../components/Breadcrumbs';
import { css } from 'glamor';

describe('Breadcrumbs', () => {
  it('renders with props', () => {
    const {asFragment} = render(
      <MemoryRouter initialEntries={['/vacolsId/documents/docId']}>
        <Breadcrumbs>
          <div breadcrumb="Document Viewer" path="/:vacolsId/documents/:docId" />
          <div breadcrumb="Claims Folder" path="/:vacolsId/documents" />
          <div path="/" title="Assignments | Caseflow Reader" />
        </Breadcrumbs>
      </MemoryRouter>
    );
  
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders array of crumbs, custom label fn, styling, without initial caret', () => {
    const {asFragment} = render(
      <MemoryRouter initialEntries={['/vacolsId/documents/docId']}>
        <Breadcrumbs
          getBreadcrumbLabel={(route) => route.breadcrumb}
          shouldDrawCaretBeforeFirstCrumb={false}
          styling={css({
            marginTop: '-1.5rem',
            marginBottom: '-1.5rem',
          })}
          elements={[
            {
              breadcrumb: 'Document Viewer',
              path: '/:vacolsId/documents/:docId',
            },
            {
              breadcrumb: 'Claims Folder',
              path: '/:vacolsId/documents',
            },
            {
              breadcrumb: 'Assignments | Caseflow Reader',
              path: '/',
            },
          ]}
        />
      </MemoryRouter>
    );
  
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders an array of crumbs regardless of route', () => {
    const {asFragment} = render(
      <MemoryRouter initialEntries={['/vacolsId/documents/docId']}>
        <Breadcrumbs
          getBreadcrumbLabel={(route) => route.breadcrumb}
          shouldDrawCaretBeforeFirstCrumb={false}
          renderAllCrumbs
          styling={css({
            marginTop: '-1.5rem',
            marginBottom: '-1.5rem',
          })}
          elements={[
            {
              breadcrumb: 'Your Queue',
              path: '/',
            },
            {
              breadcrumb: 'Vet. E Ran',
              path: '/tasks/:vacolsId',
            },
            {
              breadcrumb: 'Select Dispositions',
              path: '/tasks/:vacolsId/dispositions',
            },
            {
              breadcrumb: 'Submit Draft Decision',
              path: '/tasks/:vacolsId/submit',
            },
          ]}
        />
      </MemoryRouter>
    );
  
    expect(asFragment()).toMatchSnapshot();
  });
});
