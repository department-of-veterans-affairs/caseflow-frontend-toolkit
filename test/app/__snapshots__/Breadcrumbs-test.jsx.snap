// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs renders an array of crumbs regardless of route 1`] = `
<DocumentFragment>
  <div
    data-css-c6mfa5=""
  >
    <span>
      <a
        href="/"
      >
        Your Queue
      </a>
    </span>
    <span>
        &gt;  
      <a
        href="/tasks/:vacolsId"
      >
        Vet. E Ran
      </a>
    </span>
    <span>
        &gt;  
      <a
        href="/tasks/:vacolsId/dispositions"
      >
        Select Dispositions
      </a>
    </span>
    <span>
        &gt;  
      <a
        href="/tasks/:vacolsId/submit"
      >
        Submit Draft Decision
      </a>
    </span>
  </div>
</DocumentFragment>
`;

exports[`Breadcrumbs renders array of crumbs, custom label fn, styling, without initial caret 1`] = `
<DocumentFragment>
  <div
    data-css-c6mfa5=""
  >
    <a
      href="/"
    >
      Assignments | Caseflow Reader
    </a>
      &gt;  
    <a
      href="/vacolsId/documents"
    >
      Claims Folder
    </a>
      &gt;  
    <a
      href="/vacolsId/documents/docId"
    >
      Document Viewer
    </a>
  </div>
</DocumentFragment>
`;

exports[`Breadcrumbs renders with props 1`] = `
<DocumentFragment>
  <div
    data-css-1mdtwij=""
  >
      &gt;  
    <a
      href="/vacolsId/documents"
    >
      <h2
        class="cf-application-title"
        id="page-title"
      >
        Claims Folder
      </h2>
    </a>
      &gt;  
    <a
      href="/vacolsId/documents/docId"
    >
      <h2
        class="cf-application-title"
        id="page-title"
      >
        Document Viewer
      </h2>
    </a>
  </div>
</DocumentFragment>
`;
